includes/class-wc-calypso-bridge-admin-menu.php

```diff
@@ -10,11 +10,331 @@
+		add_submenu_page( 'options-general.php', __( 'WooCommerce Settings','wc-calypso-bridge' ), __( 'WooCommerce','wc-calypso-bridge' ), 'manage_woocommerce', 'admin.php?page=wc-settings', '', 10 );
+
+		// Move WooCommerce > Status under Tools > WooCommerce Status.
+		$this->hide_submenu_page( 'woocommerce', 'wc-status' );
+		add_submenu_page( 'tools.php', __( 'WooCommerce Status','wc-calypso-bridge' ), __( 'WooCommerce Status','wc-calypso-bridge' ), 'manage_woocommerce', 'admin.php?page=wc-status', '', 10 );
+
+		// Move legacy reports.
+		$this->hide_submenu_page( 'woocommerce', 'wc-reports' );
+		// Force-add the "Report (Legacy)" submenu item.
+		if ( ! empty( $submenu['wc-admin&path=/analytics/overview'] ) && is_array( $submenu['wc-admin&path=/analytics/overview'] ) ) {
+			$submenu['wc-admin&path=/analytics/overview'][] = array(
+				__( 'Reports (Legacy)', 'wc-calypso-bridge' ),
+				'manage_woocommerce',
+				'admin.php?page=wc-reports',
+				'Reports (Legacy)'
+			);
+		}
+
+		// Move Customers to root menu.
+		$this->hide_submenu_page( 'woocommerce', 'wc-admin&path=/customers' );
+		add_menu_page( __( 'Customers', 'woocommerce' ), __( 'Customers', 'woocommerce' ), 'manage_woocommerce', '/admin.php?page=wc-admin&path=/customers', null, 'dashicons-money', 100 );
+
+		// Update WooCommerce to Extensions
+		$this->update_menu( 'woocommerce', null, 'Extensions', null, null, null );
+
+		// Add Extensions > Manage submenu.
+		add_submenu_page( 'woocommerce', __( 'WooCommerce Subscriptions','wc-calypso-bridge' ), __( 'Manage','wc-calypso-bridge' ), 'manage_woocommerce', 'admin.php?page=wc-addons&section=helper', '', 10 );
+
+
+		// Move WooCommerce > Extensions under Extensions > Discover.
+		foreach ( $submenu['woocommerce'] as $key => $data ) {
+			if ( 'wc-addons' !== $data[2] ) {
+				continue;
+			}
+			$submenu['woocommerce'][$key][0] = __( 'Discover', 'wc-calypso-bridge' );
+		}
+
+		// Add Orders count.
+		if ( apply_filters( 'woocommerce_include_processing_order_count_in_menu', true ) && current_user_can( 'edit_others_shop_orders' ) ) {
+			$order_count = (int) apply_filters( 'woocommerce_menu_order_count', wc_processing_order_count() );
+
+			if ( $order_count ) {
+				foreach ( $menu as $i => $menu_item ) {
+					if ( 'edit.php?post_type=shop_order' === $menu_item[2] ) {
+						$menu[$i][0] .= ' <span class="awaiting-mod update-plugins count-' . esc_attr( $order_count ) . '"><span class="processing-count">' . number_format_i18n( $order_count ) . '</span></span>'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+						break;
+					}
+				}
+			}
+		}
+
+		if ( ! empty( $submenu['woocommerce'] ) && is_array( $submenu['woocommerce'] ) ) {
+
+			// Make sure that the managed and hidden are the last on the submenu list. This is make the parent "WooCommerce" item to point to an extension instead of a hidden page.
+			uasort( $submenu['woocommerce'], function( $a, $b ) {
```

Comment:
Might it be worth moving these custom sort functions to dedicated helper functions? The `add_woocommerce_menu()` function is currently over 180 lines long, with a lot of that taken up by the sort functions.